google-cloud-storage/src/main/java/com/google/cloud/storage/BucketInfo.java

```diff
@@ -799,13 +804,21 @@ public LifecycleCondition build() {
     }
 
     /**
-     * Base class for the Action to take when a Lifecycle Condition is met. Specific Actions are
+     * Base class for the Action to take when a Lifecycle Condition is met. Supported Actions are
      * expressed as subclasses of this class, accessed by static factory methods.
      */
-    public abstract static class LifecycleAction implements Serializable {
+    public static class LifecycleAction implements Serializable {
       private static final long serialVersionUID = 5801228724709173284L;
 
-      public abstract String getActionType();
+      private final String actionType;
+
+      private LifecycleAction(String actionType) {
```

Comment:
Let's make this constructor `public` to allow someone (as remote a chance as it is) that is already extending themselves allow them to continue to extend, but now they would need to provide the `actionType`.